src/core/AutomationPattern.cpp

```diff
@@ -787,6 +787,16 @@ void AutomationPattern::resolveAllIDs()
 						{
 							a->addObject( dynamic_cast<AutomatableModel *>( o ), false );
 						}
+						else
+						{
+							// FIXME: Remove this block once the automation system gets fixed
+							// This is a temporary fix for https://github.com/LMMS/lmms/issues/3781
+							o = Engine::projectJournal()->journallingObject( *k + ( 1 << 23 ) );
```

Comment:
It is dangerous to mess with ProjectJournal internals like this. You should request new ids and copy data if necessary.

Reply:
I also think this is a bad design, but this is the minimal fix for the bug which doesn't involve possibly dangerous structure changes for stable releases.
I want to remove this one in `master` as soon as I get my work completed.

Reply:
Please make it less minimal. Do not expose `EO_ID_MSB`.

Reply:
I didn't want to do so. However, changing `ProjectJournal` may affect other stuff than automation. I can address this, but it will make the *temporary* fix much complicated.

Reply:
Please address this.

Reply:
Do you want to become a coder or not?

Reply:
What do you mean?

Reply:
Let us assume you want to be a better coder. Minimizing structural changes does not mean to make the smallest patch. Minimizing changes certainly does not mean breaking code organization.

I have asked you an easy task: do not expose `EO_ID_MSB`. It will not make your fix that much complicated. Could you give it a try? Do you need a hint?

Reply:
I agree that the code organization is important, but we're on a branch targetting an upcoming release. Code organization is important in the long term, but bug fixes are more important for now. We consider design changes are out of the scope of `stable-1.2`, though they are important in `master`.
I will remove the exposure of `EO_ID_MSB` in `master`, with new changes in the design.

Reply:
Fine, I will fix your fix.

Reply:
On which branch, and how?

Reply:
@PhysSong I think he intends fixing this in stable-1.2.
Merge this PR and we'll know for sure.

Reply:
Okay. BTW, @jasp00, please note that you will probably make double work. I already have a fix with new designs and what I need to do is just cleaning them up.

Reply:
I think it makes more sense to argue using comments, rather than to argue using commits. We can merge or close once we've reached a consensus on what the code should look like.
Is the desired change something as simple as, e.g.
```cpp
jo_id_t ProjectJournal::idFromSave( jo_id_t id )
{
	return id | EO_ID_MSB;
}
```
?

Reply:
I'm not sure if we need that, but that could be a solution.

Reply:
@jasp00 If you're comfortable with the @DomClark's suggestion, I will use that for now.

Reply:
OK.